optimum/onnxruntime/optimization.py

```diff
@@ -189,15 +213,25 @@ def optimize(
 
             suffix = f"_{file_suffix}" if file_suffix else ""
             output_path = save_dir.joinpath(f"{model_path.stem}{suffix}").with_suffix(model_path.suffix)
-            optimizer.save_model_to_file(output_path.as_posix(), use_external_data_format, one_external_file)
+
+            # TODO: ORT save_model_to_file will save as `.data` although we save as `.onnx_data` in the export
+            optimizer.save_model_to_file(
+                output_path.as_posix(),
+                use_external_data_format=model_uses_external_data,
+                all_tensors_to_one_file=one_external_file,
+            )
+
+            # if loading from disk and saving in the same repository, remove previous external data
+            if Path(model_path.as_posix() + "_data").is_file() and self.from_ortmodel is False:
```

Comment:
The second part of the if statement is ugly IMO.

Also it does not read the same way, why not doing that then if the plan is to be very verbose?
```python
if Path(model_path.as_posix() + "_data").is_file() is True and self.from_ortmodel is False:
```

Reply:
Personally I'm in favor of a more pythonic way:
```python
if Path(model_path.as_posix() + "_data").is_file() is True and not self.from_ortmodel:
```

Reply:
`is_file()` is defined in pathlib, and I expect pathlib to respect its signatures given in documentation. `self.from_ortmodel` may be passed / changed by the user/ourselves, and I don't expect the user to respect always signatures.

Reply:
Then I would also check that `model_path` is actually a `Path`, not ` os.PathLike` as annotated in the `__init__ before trying `model_path.as_posix()` because the user can pass this value. 

Anyway, just nitpicking but I truly think it's not readable. Feel free to ignore ;)